src/arrayMethodSort.js

```diff
@@ -4,8 +4,48 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompareFunction = (a, b) => {
+      const firstElement = String(a);
+      const secondElement = String(b);
+
+      if (firstElement > secondElement) {
+        return 1;
+      } else if (firstElement < secondElement) {
+        return -1;
+      }
+
+      return 0;
+    };
+
+    function quickSort(arr, compare) {
+      if (arr.length <= 1) {
+        return arr;
+      }
+
+      const pivot = arr[arr.length - 1];
+      const left = [];
+      const right = [];
+
+      for (let i = 0; i < arr.length - 1; i++) {
+        if (compareFunc(arr[i], pivot) < 0) {
```

Comment:
There is a typo in the variable name `compareFunc`. It should match the parameter name `compare` defined in the `quickSort` function signature.